foreign_cc/private/framework/platform.bzl

```diff
@@ -92,10 +152,37 @@ def target_os_name(ctx):
     Returns:
         str: The string of the current platform
     """
-    operating_systems = ["android", "linux"]
-    for os in operating_systems:
+    for os in SUPPORTED_OS:
         constraint = getattr(ctx.attr, "_{}_constraint".format(os))
         if constraint and ctx.target_platform_has_constraint(constraint[platform_common.ConstraintValueInfo]):
             return os
 
     return "unknown"
+
+def triplet_name(os, arch):
+    """A helper function for getting the platform triplet from the results of the above arch/os functions
+
+    Args:
+        os (str): the os
+        arch (str): the arch
+
+    Returns:
+        str: The string of the current platform
+    """
+
+    # This is like a simplified config.guess / config.sub from autotools
+    if os == "linux":
+        if arch == "aarch64":
+            return "aarch64-unknown-linux-gnu"
+        elif arch == "s390x":
+            return "s390x-ibm-linux-gnu"
+        elif arch == "x86_64":
+            return "aarch64-pc-linux-gnu"
+
+    elif os == "macos":
+        if arch == "aarch64":
+            return "aarch64-apple-macos"
+        elif arch == "x86_64":
+            return "x86_64-apple-macos"
```

Comment:
Is macos correct? On my mac at least the triplet names used for gcc installed via homebrew are aarch64-apple-darwin23

Reply:
Yeah, the convention for homebrew is `${arch}-apple-darwin${version}`. Other examples we saw internally (from running config.guess on those platforms) were (aarch64|x86_64)-apple-darwin23.(4|5).0 and I was slightly nervous about lying about the library version, so I checked to see what config.sub would do (since that's what autotools uses to normalize) and it uses `macos` as the default os name, but then also preserves the `darwinXX` when it sees it: 
```
$ ./config.sub-63acb96f92473ceb5e21d873d7c0aee266b3d6d3 aarch64-apple-macos
aarch64-apple-macos
$ ./config.sub-63acb96f92473ceb5e21d873d7c0aee266b3d6d3 aarch64-apple
aarch64-apple-macos
$ ./config.sub-63acb96f92473ceb5e21d873d7c0aee266b3d6d3 aarch64-apple-darwin123
aarch64-apple-darwin123
```

I took this as a sign that macos was fine. A slightly more correct way to guess the host might actually be to ship config.guess and run it as an action to see what it says, and then use that, but since there's no good way to do that reliably for the target tuple, I biased towards internally-consistent normalization.

One _other_ consideration is, I don't currently pass `--build`, just `--host`. This means it's possible that  autoconf's cross_compiling flag will get set to true when it shouldn't be (like in the darwin case above). I did that because it's more likely to result in a working result when the target's not in our tuple, but it does also mean passing `--host aarch64-apple-darwin23` when config.guess thinks it's `aarch64-apple-darwin23.4` triggers cross-compiling mode. The fix for this would be to pass both `--host` and `--build` all the time, so our consistent normalization prevents that from being a problem. (I did confirm this in our CI, our darwin builds do end up emitting `checking whether we are cross compiling... yes`)

I'm happy to adjust this to whichever variation you think is best! After having written this out, I'm actually more inclined to pass --build under every scenario we pass --host and rely on other users updating the data as-needed, but I'll defer to you.

Reply:
I did think of a disadvantage of passing --build: there's no way (that I know of) for bazel to guess which libc the compiler is expecting, because there aren't consistent libc constraints to use. I don't think this actually breaks anything, though, because it appears that when autoconf finds musl it actually internally replaces the triplet with gnu anyway.

I think I'll make the change anyway (it can't break any _existing_ code because autotools would already be breaking in that case) but just something to keep an eye on.